config/config.default.php

```diff
@@ -7,96 +7,86 @@
  */
 
 return array(
-    // Which backend to use for Xhgui_Saver.
-    // Must be one of 'mongodb', or 'file'.
-    //
-    // Example (save to a temporary file):
-    //
-    //     'save.handler' => 'file',
-    //     # Beware of file locking. You can adujst this file path
-    //     # to reduce locking problems (eg uniqid, time ...)
-    //     'save.handler.filename' => __DIR__.'/../data/xhgui_'.date('Ymd').'.dat',
-    //
-    'save.handler' => 'mongodb',
+    'debug'     => false,
+    'mode'      => 'development',
 
-    'pdo' => array(
-        'dsn' => null,
-        'user' => null,
-        'pass' => null,
-        'table' => 'results'
-    ),
+    // Can be mongodb, file, upload or pdo.
 
-    // Database options for MongoDB.
-    //
-    // - db.host: Connection string in the form `mongodb://[ip or host]:[port]`.
-    //
-    // - db.db: The database name.
+    // For file
+    'save.handler'                    => 'file',
+    'save.handler.filename'           => dirname(__DIR__) . '/cache/' . 'xhgui.data.' . microtime(true) . '_' . substr(md5(!empty($_SERVER['REQUEST_URI']) ? $_SERVER['REQUEST_URI'] : microtime(true).getmypid().uniqid('fallback_requiest_id', true)), 0, 6),
```

Comment:
Using `random_bytes()` should suffice here, and might be easier to reason about.

Pulling it through `md5()` defeats most of the entropy that goes in, so instead of that, one could use `bin2hex()`, like so:

```php
microtime(true) . '_' . bin2hex(random_bytes(3));
```

Which produces a stronger rand, in the same ~ 6 character space ([example](https://3v4l.org/4riR9)).